src/NHibernate.Test/Linq/ProjectionsTests.cs

```diff
@@ -275,6 +340,18 @@ public void ProjectAnonymousTypeWithCollection3()
 						select new { OrderLines = o.OrderLines.ToList() };
 
 			var result = query.ToList();
+			Assert.That(NHibernateUtil.IsInitialized(result[0].OrderLines), Is.True);
+			Assert.That(result.Count, Is.EqualTo(830));
+		}
+
+		[Test]
+		public void ProjectAnonymousTypeWithCollection3WithFetch()
+		{
+			var query = from o in db.Orders.Fetch(o => o.OrderLines)
+						select new { OrderLines = o.OrderLines.ToList() };
```

Comment:
The sql created from this query returns more than 7000 rows, I think that it would be better to use the `PatientRecords` collection of `Patient` which has less data:
```
var query = from o in db.Patients.Fetch(o => o.PatientRecords)
						select new { PatientRecords = o.PatientRecords.ToList() };
```
the same applies for other queries that were added by this PR.

Reply:
Perhaps I should just add the test from the original bug report. Since doing a Fetch when projecting a collection is actually redundant. Adding that one test will guard against a regression for this code.

Reply:
> Perhaps I should just add the test from the original bug report

Fine by me

Reply:
I am also ok with that.